src/dosext/mfs/mfs.c

```diff
@@ -2633,12 +2692,8 @@ static void RemoveRedirection(int drive, cds_t cds)
   cds_rootlen(cds) = CDS_DEFAULT_ROOT_LEN;
   cds_cur_cluster(cds) = 0;	/* reset us at the root of the drive */
 
-  /* see if there is a physical drive behind this redirection */
-  DPBptr = cds_DPB_pointer(cds);
-  if (DPBptr.offset | DPBptr.segment) {
-    /* if DPB_pointer is non-NULL, set the drive status to ready */
+  if (hasPhysical(cds))
     cds_flags(cds) = CDS_FLAG_READY;
```

Comment:
Misindented.
And what about FLAG_NONNET?

Reply:
> Misindented.

I don't believe so.

CDS_FLAG_NOTNET is normally used with CDS_FLAG_REMOTE to indicate cdrom, I've never seen it used with CDS_FLAG_READY

Reply:
Ahh, I see your point.
However, please take a look here:
http://faydoc.tripod.com/structures/16/1643.htm
```
bit 15 | : | uses network redirector | invalid if 00, installable file system if 11
-- | -- | -- | --
bit 14 | : | physical drive
bit 13 | : | JOIN'ed | path above is true path that would be needed if not         under SUBST or JOIN
bit 12 | : | SUBST'ed
bit 11 | : | removable media (WinNT4.0)
bit 10 | : | hard disk??? (WinNT4.0)
bit 7 | : | remote drive hidden from redirector's assign-list and         exempt from network connection make/break commands; set for CD-ROM         drives by MSCDEX (not supported by CORELCDX)
```

So `READY` is actually "physical drive" and
"NOTNET" is "cdrom". If we rename them that way,
there would be no need to alter "physical drive" flag -
it will always be valid, regardless of the redirection.
What do you think?

Reply:
Depends upon what DOS does if both PHYSICAL and REMOTE are set. I can imagine it breaking at some DOS variants and I expect other redirectors don't allow redirection over a physical drive so I doubt it's been tested.

Reply:
> Depends upon what DOS does if both PHYSICAL and REMOTE are set.

But that's the point!
We already do:
`cds_flags(cds) = CDS_FLAG_READY | CDS_FLAG_REMOTE | CDS_FLAG_NOTNET;`
which sets both.
So what do you think, taking that into account?